ext/test/http/CMakeLists.txt

```diff
@@ -25,7 +25,7 @@ endif()
 set(URL_PARSER_FILENAME url_parser_test)
 add_executable(${URL_PARSER_FILENAME} ${URL_PARSER_FILENAME}.cc)
 target_link_libraries(${URL_PARSER_FILENAME} ${GTEST_BOTH_LIBRARIES}
-                      ${CMAKE_THREAD_LIBS_INIT})
+                      ${CMAKE_THREAD_LIBS_INIT} opentelemetry_api)
```

Comment:
Is this required?

Reply:
I think it's required. It will all the public definitions of `opentelemetry_api` to `${URL_PARSER_FILENAME}`.

Reply:
It is required for unit test url_parser_test to honor WITH_STL, see the PR comments for details.